<?xml version='1.1' encoding='UTF-8'?>
<Tag plugin="workflow-support@3.4">
  <node class="cps.n.StepAtomNode" plugin="workflow-cps@2.80">
    <parentIds>
      <string>11</string>
    </parentIds>
    <id>12</id>
    <descriptorId>org.jenkinsci.plugins.workflow.steps.durable_task.ShellStep</descriptorId>
  </node>
  <actions>
    <cps.a.ArgumentsActionImpl plugin="workflow-cps@2.80">
      <arguments>
        <entry>
          <string>script</string>
          <string>
            #!/usr/bin/env bash
            #aws s3 ls
            #grep ${repoName}
            aws s3 ls | grep ${repoName} || true
            status=(&quot;${PIPESTATUS[@]}&quot;)
            AWSCODE=${status[0]}
            GREPCODE=${status[1]}
            echo $AWSCODE $GREPODE
            if [ $GREPCODE -eq 0 ]
            then
                # --delete : for deleting any files that are exist in source and not in S3
                aws s3 sync . s3://${repoName}-${repoBranch} --exclude &apos;.git/*&apos; --delete
                echo &quot;Finished creation of the repo to S3 Bucket Name : ${repoName}-${repoBranch}&quot;
            else
                aws s3 mb s3://${repoName}-${repoBranch}
                aws s3 cp . s3://${repoName}-${repoBranch} --recursive --exclude &apos;.git/*&apos;
                echo &quot;Finished upload the repo to S3 Bucket Name : ${repoName}-${repoBranch}&quot;
            fi
        </string>
        </entry>
        <entry>
          <string>label</string>
          <string>Checking if S3 exists</string>
        </entry>
      </arguments>
      <isUnmodifiedBySanitization>true</isUnmodifiedBySanitization>
    </cps.a.ArgumentsActionImpl>
    <wf.a.TimingAction plugin="workflow-api@2.40">
      <startTime>1588154338328</startTime>
    </wf.a.TimingAction>
    <s.a.LogStorageAction/>
    <wf.a.LabelAction plugin="workflow-api@2.40">
      <displayName>Checking if S3 exists</displayName>
    </wf.a.LabelAction>
    <wf.a.ErrorAction plugin="workflow-api@2.40">
      <error class="hudson.AbortException">
        <detailMessage>script returned exit code 2</detailMessage>
        <stackTrace>
          <trace>org.jenkinsci.plugins.workflow.steps.durable_task.DurableTaskStep$Execution.handleExit(DurableTaskStep.java:658)</trace>
          <trace>org.jenkinsci.plugins.workflow.steps.durable_task.DurableTaskStep$Execution.check(DurableTaskStep.java:604)</trace>
          <trace>org.jenkinsci.plugins.workflow.steps.durable_task.DurableTaskStep$Execution.run(DurableTaskStep.java:548)</trace>
          <trace>java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)</trace>
          <trace>java.util.concurrent.FutureTask.run(FutureTask.java:266)</trace>
          <trace>java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)</trace>
          <trace>java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)</trace>
          <trace>java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)</trace>
          <trace>java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)</trace>
          <trace>java.lang.Thread.run(Thread.java:748)</trace>
        </stackTrace>
        <suppressedExceptions class="java.util.Collections$UnmodifiableRandomAccessList" resolves-to="java.util.Collections$UnmodifiableList">
          <c class="list"/>
          <list reference="../c"/>
        </suppressedExceptions>
      </error>
    </wf.a.ErrorAction>
  </actions>
</Tag>